packages/@aws-cdk/aws-ec2-alpha/lib/route.ts

```diff
@@ -402,6 +436,106 @@ export class NatGateway extends Resource implements IRouteTarget {
   }
 }
 
+/**
+ * Creates a peering connection between two VPCs
+ * @resource AWS::EC2::VPCPeeringConnection
+ */
+export class VPCPeeringConnection extends Resource implements IRouteTarget {
+
+  /**
+   * The type of router used in the route.
+   */
+  readonly routerType: RouterType;
+
+  /**
+   * The ID of the route target.
+   */
+  readonly routerTargetId: string;
+
+  /**
+   * The VPC peering connection CFN resource.
+   */
+  public readonly resource: CfnVPCPeeringConnection;
+
+  constructor(scope: Construct, id: string, props: VPCPeeringConnectionProps) {
+    super(scope, id);
+
+    this.routerType = RouterType.VPC_PEERING_CONNECTION;
+
+    const isCrossAccount = props.requestorVpc.ownerAccountId !== props.acceptorVpc.ownerAccountId;
+
+    if (!isCrossAccount && props.peerRoleArn) {
+      Annotations.of(scope).addWarningV2(
+        '@aws-cdk/aws-ec2-alpha:peerRoleArnIgnored',
+        'This is a same account peering, peerRoleArn is not needed and will be ignored',
```

Comment:
why do you raise warning, and not an Exception?

Reply:
We can still establish a same account peering connection even if they input peerRoleArn as a field it would just be ignored. I didn't think it needed an exception since the connection could still be established. What do you think?

Reply:
IMO, it is better to raise an exception, and let the customers know that they are doing something wrong.

Reply:
sounds good! [b8fc4a0](https://github.com/aws/aws-cdk/pull/31645/commits/b8fc4a000c946e12ecfe9e5310cf3e8b668070a3)